src/font_face.rs

```diff
@@ -270,6 +272,43 @@ impl FontFace {
             (*self.native.get()).GetIndex()
         }
     }
+
+    pub fn has_variations(&self) -> bool {
+        unsafe {
+            let face5: Option<ComPtr<IDWriteFontFace5>> = (*self.native.get()).query_interface(&IDWriteFontFace5::uuidof());
+            if let Some(face) = face5 {
+                return face.HasVariations() == TRUE;
+            }
+            false
+        }
+    }
+
+    pub fn create_font_face_with_variations(
+        &self,
+        simulations: DWRITE_FONT_SIMULATIONS,
+        axis_values: &[DWRITE_FONT_AXIS_VALUE],
+    ) -> Option<FontFace> {
+        unsafe {
+            let face5: Option<ComPtr<IDWriteFontFace5>> = (*self.native.get()).query_interface(&IDWriteFontFace5::uuidof());
```

Comment:
is this how extension methods are discovered in dwrote? I wonder if the query_interface would be better to call at the construction site and just cache this `ComPtr<IDWriteFontFace5>` somewhere.